content/en/blog/2022-03-09-pod-security-standard-restricted.md

```diff
@@ -0,0 +1,125 @@
+> the container isolation. The recently disclosed [kernel vulnerability
+> CVE-2022-0185](https://blog.aquasec.com/cve-2022-0185-linux-kernel-container-escape-in-kubernetes)
+> is a good example of that.
+
+## Pod Security Standards definition
+
+Kubernetes defined three policies in its Pod Security Standards. They
+range from
+
+- **Privileged**: This does not place any restrictions on the workload
+  at all. The idea being that this can be used for system- and
+  infrastructure-level workloads which are managed by privileged and
+  trusted users only.
+- **Baseline**: This policy comes with some restrictions. It aims to
+  guard against known privilege escalations while still making it
+  easy to adopt and use it by keeping a certain level of
+  compatibility with most workloads.
+- **Restricted**: Inheriting all the restrictions from *Baseline*, it
+  enforces additional security and thus follows hardening best
+  practices.
+
+We are very pleased that all Flux controllers were moved to
+*Restricted*, as that offers the highest level of security for you.
+
+We recommend checking out the [Upstream Kubernetes documentation on Pod
+Security
+Standards](https://kubernetes.io/docs/concepts/security/pod-security-standards/)
+as it gives a generally good overview of all the security features
+enabled. In addition to that you can see which restrictions were added
+as part of which Kubernetes release, meaning that with every Kubernetes
+release, you will benefit from new Upstream Kubernetes security
+improvements automatically.
+
+## `RuntimeDefault` and `seccomp`
+
+Seccomp is short for "Secure Computing". It refers to a facility in the
+Linux kernel which can limit the number of system calls available to a
+given process. Right now there are around 300+ system calls available,
+e.g. `read` to read from a file descriptor or `chmod` to change the
+permissions of a file. The more syscalls you block, the more secure your
+application, as a rogue process will only be able to do what you
+specified.
+
+In its first inception `seccomp` was introduced into Linux in 2005, to
+Docker in version 1.10 (Feb 2016) and to Kubernetes in version 1.3 (Jul
+2016). So while the technology has been around for a while and you could
+handcraft your own `seccomp` profiles, the challenge has always been
+striking the right balance: if you are too generous in your filter, it
+won't guard against malware effectively -- if you are too strict, your
+application might not work.
+
+All container runtimes come with a default seccomp profile. [Docker
+Desktop for
+example](https://github.com/moby/moby/blob/master/profiles/seccomp/default.json)
+blocks around 44 system calls. In Kubernetes you can enable the seccomp
+profile RuntimeDefault for your pod like so:
+
+```yaml
+spec:
+  securityContext:
+    seccompProfile:
+      type: RuntimeDefault
+```
+
+All Flux controllers have this implemented as well now! This is just one
+more measure we are taking to keep you more secure.
+
```

Comment:
Wrapping up we could just summarise that by adopting both changes, we further restrict the permissions that flux requires in order to operate. This, alongside other changes we are working on, translates in a decreased attack surface which may reduce the impact of eventual CVEs that may surfaced in our code base - or our supply chain.